File: project/utils/cashfree.py
import aiohttp
from project.utils.vars import Var

session: aiohttp.ClientSession = None

async def init_session():
    global session
    headers = {
        'x-client-id': Var.CF_CLIENTID,
        'x-client-secret': Var.CF_CLIENTSECRET,
        'Accept': 'application/json',
        'Content-Type': 'application/json',
        'x-api-version': Var.CF_VERSION
    }
    session = aiohttp.ClientSession(
        "https://sandbox.cashfree.com", headers=headers)

async def create_order(admid: str, phone: str, name: str, email: str, uuid4: str, amount: int, return_path: str):
    global session
    if not session:
        await init_session()
    data = {
        "order_amount": amount,
        "order_currency": "INR",
        "order_id": uuid4,
        "customer_details": {
            "customer_id": admid,
            "customer_phone": phone,
            "customer_name": name,
            "customer_email": email
        },
        "order_meta": {
            "return_url": Var.URL+return_path+"?order_id={order_id}",
            "payment_methods": "cc,dc,upi"
        }
    }

    async with session.post('/pg/orders', json=data) as resp:
        resp_data=await resp.json()
        return resp_data.get("payment_session_id")

async def fetch_payment(order_id: str):
    global session
    if not session:
        await init_session()
    async with session.get(f'/pg/orders/{order_id}/payments') as resp:
        resp_data=await resp.json()
        return resp_data

File: utils/cashfree.php
<?php
require_once $_SERVER['DOCUMENT_ROOT'] . '/utils/vars.php';


function createOrder($admid, $phoneno, $name, $email, $uuid4, $amount)
{
    $url = "https://sandbox.cashfree.com/pg/orders"; // Update to production URL for live
    $x_client_id = Config::CF_CLIENTID; // Replace with your Cashfree client ID
    $x_client_secret = Config::CF_CLIENTSECRET; // Replace with your Cashfree client secret
    $x_api_version = "2023-08-01";

    // Prepare request data
    $postData = [
        "order_id" => $uuid4,
        "order_amount" => $amount,
        "order_currency" => "INR",
        "customer_details" => [
            "customer_id" => "$admid",
            "customer_email" => $email,
            "customer_phone" => $phoneno,
            "customer_name" => $name,
        ],
        "order_meta" => [
            "return_url" => Config::URL."/student/order/checkout.php?order_id={order_id}",
            "payment_methods" => "cc,dc,upi"
        ]
    ];

    // Initialize cURL
    $ch = curl_init($url);

    // Set cURL options
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
    curl_setopt($ch, CURLOPT_POST, true);
    curl_setopt($ch, CURLOPT_POSTFIELDS, json_encode($postData));
    curl_setopt($ch, CURLOPT_HTTPHEADER, [
        "Content-Type: application/json",
        "x-api-version: $x_api_version",
        "x-client-id: $x_client_id",
        "x-client-secret: $x_client_secret"
    ]);

    // Execute the request
    $response = curl_exec($ch);

    // Check for errors
    if (curl_errno($ch)) {
        error_log('cURL error: ' . curl_error($ch));
        return null;
    }

    // Close cURL
    curl_close($ch);

    // Decode the JSON response
    $responseData = json_decode($response, true);

    // Check for the payment_session_id and return it
    if (isset($responseData['payment_session_id'])) {
        return $responseData['payment_session_id'];
    } else {
        error_log('Error in response: ' . $response);
        return null;
    }
}

function fetchPayment($order_id){
    $url = "https://sandbox.cashfree.com/pg/orders/$order_id/payments";
    $x_client_id = Config::CF_CLIENTID;
    $x_client_secret = Config::CF_CLIENTSECRET;
    $x_api_version = "2023-08-01";

    // Initialize cURL
    $ch = curl_init($url);

    // Set cURL options
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
    curl_setopt($ch, CURLOPT_HTTPHEADER, [
        "x-client-id: $x_client_id",
        "x-client-secret: $x_client_secret",
        "Accept: application/json",
        "x-api-version: $x_api_version"
    ]);

    $response = curl_exec($ch);

    if (curl_errno($ch)) {
        error_log('cURL error: ' . curl_error($ch));
        return null;
    }
    curl_close($ch);

    $responseData = json_decode($response, true);

    if (is_array($responseData) && isset($responseData[0])) {
        return $responseData[0];
    } else {
        error_log('Invalid response format: ' . $response);
        return null;
    }
}
?>